examples/pytorch/grand/main.py

```diff
@@ -0,0 +1,198 @@
+    parser.add_argument('--lr', type=float, default=0.01, help='Learning rate.')
+    parser.add_argument('--weight_decay', type=float, default=5e-4, help='L2 reg.')
+    # model params
+    parser.add_argument("--hid_dim", type=int, default=32, help='Hidden layer dimensionalities.')
+    parser.add_argument('--dropnode_rate', type=float, default=0.5,
+                        help='Dropnode rate (1 - keep probability).')
+    parser.add_argument('--input_droprate', type=float, default=0.0,
+                    help='dropout rate of input layer')
+    parser.add_argument('--hidden_droprate', type=float, default=0.0,
+                    help='dropout rate of hidden layer')
+    parser.add_argument('--order', type=int, default=8, help='Propagation step')
+    parser.add_argument('--sample', type=int, default=4, help='Sampling times of dropnode')
+    parser.add_argument('--tem', type=float, default=0.5, help='Sharpening temperature')
+    parser.add_argument('--lam', type=float, default=1., help='Lamda')
+    parser.add_argument('--use_bn', action='store_true', default=False, help='Using Batch Normalization')
+
+    args = parser.parse_args()
+    if args.gpu != -1 and th.cuda.is_available():
+        args.device = 'cuda:%s'%args.gpu
+    else:
+        args.device = 'cpu'
+
+    return args
+
+if __name__ == '__main__':
+
+    args = argument()
+    print(args)
+    if args.dataname == 'cora':
+        dataset = CoraGraphDataset()
+    elif args.dataname == 'citeseer':
+        dataset = CiteseerGraphDataset()
+    elif args.dataname == 'pubmed':
+        dataset = PubmedGraphDataset()
+        
+    graph = dataset[0]
+    
+    graph = dgl.add_self_loop(graph)
+    device = args.device
+
+    n_classes = dataset.num_classes
+
+    labels = graph.ndata.pop('label').to(device).long()
+    feats = graph.ndata.pop('feat').to(device)
+    
+    n_features = feats.shape[-1]
+
+    train_mask = graph.ndata.pop('train_mask')
+    val_mask = graph.ndata.pop('val_mask')
+    test_mask = graph.ndata.pop('test_mask')
+
+    train_idx = th.nonzero(train_mask, as_tuple=False).squeeze().to(device)
+    val_idx = th.nonzero(val_mask, as_tuple=False).squeeze().to(device)
+    test_idx = th.nonzero(test_mask, as_tuple=False).squeeze().to(device)
+
+    
+    in_dim = n_features
+    hid_dim = args.hid_dim
+    n_class = n_classes
+    K = args.order
+    S = args.sample
+    T = args.tem
+    lam = args.lam
+    batchnorm = args.use_bn
+    
+    node_dropout = args.dropnode_rate
+    input_droprate = args.input_droprate
+    hidden_droprate = args.hidden_droprate
+
+    model = GRAND(in_dim, hid_dim, n_class, S, K,
+                  node_dropout, input_droprate, 
+                  hidden_droprate, batchnorm)
+
+    
+    model = model.to(device)
+    graph = graph.to(device)
+    
+
+    loss_fn = nn.NLLLoss()
+    opt = optim.Adam(model.parameters(), lr = args.lr, weight_decay = args.weight_decay)
```

Comment:
Suggest to either use all argument as args.*, or assign argument as in line 107 to 118.